contrib/translations/PDE-reduction and translations.ipynb

```diff
@@ -2,239 +2,185 @@
+    "\n",
     "        for pde_coeff, coeff_id_offset in pde:\n",
     "            other_coeff = add_tuples(base_coeff_id, coeff_id_offset)\n",
-    "            if not other_coeff in id_to_index:\n",
+    "            if other_coeff not in id_to_index:\n",
     "                valid = False\n",
     "                break\n",
-    "                \n",
+    "\n",
     "            pde_mat[row, id_to_index[other_coeff]] = pde_coeff\n",
-    "                \n",
+    "\n",
     "        if valid:\n",
     "            row += 1\n",
     "        else:\n",
     "            pde_mat[row] = 0\n",
-    "            \n",
+    "\n",
     "    return pde_mat[:row]\n",
     "\n",
+    "\n",
     "pde_mat = build_pde_mat(mpole_expn, pde)"
    ]
   },
   {
    "cell_type": "code",
-   "execution_count": 39,
+   "execution_count": null,
    "metadata": {},
    "outputs": [],
    "source": [
     "def find_nullspace(mat, tol=1e-10):\n",
-    "    u, sig, vt = la.svd(pde_mat, full_matrices=True)\n",
+    "    _u, sig, vt = la.svd(pde_mat, full_matrices=True)\n",
     "    zerosig = np.where(np.abs(sig) < tol)[0]\n",
-    "    if zerosig:\n",
+    "    if zerosig.size:\n",
     "        nullsp_start = zerosig[0]\n",
     "        assert np.array_equal(zerosig, np.arange(nullsp_start, pde_mat.shape[1]))\n",
     "    else:\n",
     "        nullsp_start = pde_mat.shape[0]\n",
-    "                              \n",
+    "\n",
     "    return vt[nullsp_start:].T\n",
-    "    \n",
+    "\n",
+    "\n",
     "nullsp = find_nullspace(pde_mat)"
    ]
   },
   {
    "cell_type": "code",
-   "execution_count": 40,
+   "execution_count": null,
    "metadata": {},
-   "outputs": [
-    {
-     "data": {
-      "text/plain": [
-       "4.3183836498795062e-16"
-      ]
-     },
-     "execution_count": 40,
-     "metadata": {},
-     "output_type": "execute_result"
-    }
-   ],
+   "outputs": [],
    "source": [
     "la.norm(pde_mat @ nullsp)"
    ]
   },
   {
    "cell_type": "code",
-   "execution_count": 41,
+   "execution_count": null,
    "metadata": {},
    "outputs": [],
    "source": [
     "def build_translation_mat(mexp, to_center):\n",
     "    n = len(mexp.coeffs)\n",
     "    result = np.zeros((n, n))\n",
-    "    \n",
+    "\n",
     "    for j in range(n):\n",
     "        unit_coeffs = np.zeros(n)\n",
     "        unit_coeffs[j] = 1\n",
     "        unit_mexp = mexp.with_coeffs(unit_coeffs)\n",
-    "        \n",
+    "\n",
     "        result[:, j] = t.multipole_expand(unit_mexp, to_center).coeffs\n",
-    "        \n",
+    "\n",
     "    return result\n",
     "\n",
+    "\n",
     "new_center = np.array([0, 0.5])\n",
     "tmat = build_translation_mat(mexp, new_center)"
    ]
   },
   {
    "cell_type": "code",
-   "execution_count": 42,
+   "execution_count": null,
    "metadata": {},
-   "outputs": [
-    {
-     "data": {
-      "text/plain": [
-       "<matplotlib.image.AxesImage at 0x7f7f8841ceb8>"
-      ]
-     },
```

Comment:
Was some of this output left here on purpose? This clears all the cells in both notebooks.

Reply:
No, I think it can go.